lib/Service/CapabilitiesService.php

```diff
@@ -8,52 +8,42 @@
 
 use OCA\Richdocuments\AppInfo\Application;
 use OCP\App\IAppManager;
+use OCP\Files\AppData\IAppDataFactory;
+use OCP\Http\Client\IClient;
 use OCP\Http\Client\IClientService;
-use OCP\ICache;
+use OCP\IAppConfig;
 use OCP\ICacheFactory;
 use OCP\IConfig;
 use OCP\IL10N;
 use Psr\Log\LoggerInterface;
 
-class CapabilitiesService {
-	/** @var IConfig */
-	private $config;
-	/** @var IClientService */
-	private $clientService;
-	/** @var ICache */
-	private $cache;
-	/** @var IAppManager */
-	private $appManager;
-	/** @var IL10N */
-	private $l10n;
-	/** @var LoggerInterface */
-	private $logger;
-
-	/** @var array */
-	private $capabilities;
-
-
-	public function __construct(IConfig $config, IClientService $clientService, ICacheFactory $cacheFactory, IAppManager $appManager, IL10N $l10n, LoggerInterface $logger) {
-		$this->config = $config;
-		$this->clientService = $clientService;
-		$this->cache = $cacheFactory->createDistributed('richdocuments');
-		$this->appManager = $appManager;
-		$this->l10n = $l10n;
-		$this->logger = $logger;
+class CapabilitiesService extends CachedRequestService {
+
+	private ?array $capabilities = null;
+
+	public function __construct(
+		private IClientService $clientService,
+		private ICacheFactory $cacheFactory,
+		private IAppDataFactory $appDataFactory,
+		private IAppConfig $appConfig,
+		private LoggerInterface $logger,
+		private IConfig $config,
+		private IAppManager $appManager,
+		private IL10N $l10n,
+	) {
+		parent::__construct(
+			$this->clientService,
+			$this->cacheFactory,
+			$this->appDataFactory,
+			$this->appConfig,
+			$this->logger,
+			'capabilities',
+		);
 	}
 
 	public function getCapabilities() {
 		if (!$this->capabilities) {
-			$this->capabilities = $this->cache->get('capabilities');
-		}
-
-		$isARM64 = php_uname('m') === 'aarch64';
-		$CODEAppID = $isARM64 ? 'richdocumentscode_arm64' : 'richdocumentscode';
-		$isCODEInstalled = $this->appManager->isEnabledForUser($CODEAppID);
-		$isCODEEnabled = strpos($this->config->getAppValue('richdocuments', 'wopi_url'), 'proxy.php?req=') !== false;
-		$shouldRecheckCODECapabilities = $isCODEInstalled && $isCODEEnabled && ($this->capabilities === null || count($this->capabilities) === 0);
-		if ($this->capabilities === null || $shouldRecheckCODECapabilities) {
```

Comment:
This is what has actually caused the issue, as the cache was only there for an hour every request after it expired triggered a new request to the capabilities endpoint, which was unreachable